### PR TITLE
Adds tests checking ability to update TUV-x conditions

### LINKED ISSUE #601: Allow updates to grid and profile data via the Python TUV-x class

### PR DESCRIPTION
**Pull request by mattldawson**
This PR adds a test that ensures that TUV-x profiles can be updated via the Python API. (The functionality was already in place to satisfy #601)

Adding this test revealed some incomplete functionality in initializing profiles that was added. Also, unrealistic test conditions were leading to zeros and NaNs in the TUV-x output. These conditions were adjusted to ensure non-negative real values for photolysis rate constants. More comprehensive testing of the calculated rates should be added, possibly as part of #603

closes #601

### CODE REVIEW

File: python/test/integration/test_tuvx.py [hunk 1]
```diff
@@ -269,6 +273,38 @@ def test_full_tuvx(monkeypatch):
...
+    # Print out any negative rates for debugging
+    for j in range(photolysis_rates.shape[1]):
+        negative_indices = np.where(photolysis_rates[:, j] < 0)[0]
+        if negative_indices.size > 0:
+            rate_name = list(tuvx.photolysis_rate_names.keys())[list(tuvx.photolysis_rate_names.values()).index(j)]
+            print(f"Photolysis rate '{rate_name}' has negative values at layers: {negative_indices} with values {photolysis_rates[negative_indices, j]}")
+        nan_indices = np.where(np.isnan(photolysis_rates[:, j]))[0]
+        if nan_indices.size > 0:
+            rate_name = list(tuvx.photolysis_rate_names.keys())[list(tuvx.photolysis_rate_names.values()).index(j)]
+            print(f"Photolysis rate '{rate_name}' has NaN values at layers: {nan_indices}")
```
[the remark this answers is not in the document]
Comment (mattldawson) [reply python/test/integration/test_tuvx.py]: removed!

File: python/musica/tuvx/profile.py [hunk 2]
```diff
@@ -36,18 +37,42 @@ def __init__(self, *, name: str, units: str, grid: Grid,
             edge_values: Optional array of values at grid edges (length num_sections + 1)
             midpoint_values: Optional array of values at grid midpoints (length num_sections)
             layer_densities: Optional array of layer densities (length num_sections)
+            calculate_layer_densities: If True, calculate layer densities from edge values
```
[the remark this answers is not in the document]
Comment (mattldawson) [reply python/musica/tuvx/profile.py]: updated

File: python/test/integration/test_tuvx.py [hunk 3]
```diff
@@ -182,13 +182,17 @@ def get_grid_map():
 
 def get_profile_map(grid_map):
     # Simple exponential profiles for testing
```
[the remark this answers is not in the document]
Comment (mattldawson) [reply python/test/integration/test_tuvx.py]: skipping this one

File: python/musica/tuvx/profile.py [hunk 4]
```diff
@@ -36,18 +37,42 @@ def __init__(self, *, name: str, units: str, grid: Grid,
...
+        if edge_values is None and midpoint_values is None:
+            raise ValueError("At least one of edge_values or midpoint_values must be provided.")
         if edge_values is not None:
             self.edge_values = edge_values
+        else:
+            edge_values = np.zeros(midpoint_values.size + 1, dtype=midpoint_values.dtype)
+            edge_values[1:-1] = 0.5 * (midpoint_values[:-1] + midpoint_values[1:])
+            # Extrapolate first and last edges
+            edge_values[0] = midpoint_values[0] - 0.5 * (midpoint_values[1] - midpoint_values[0])
+            edge_values[-1] = midpoint_values[-1] + 0.5 * (midpoint_values[-1] - midpoint_values[-2])
```
[the remark this answers is not in the document]
Comment (mattldawson) [reply python/musica/tuvx/profile.py]: I modified this to use edge values instead
